### ISSUE TITLE
A11y: Navbar dropdowns do not open with screen reader

### ISSUE THREAD

**Issue by Huzaif-nabi** [facebook/docusaurus#11501]
### Have you read the Contributing Guidelines on issues?

- [x] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [x] I'm using the latest version of Docusaurus.
- [x] I have tried the `npm run clear` or `yarn clear` command.
- [x] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [x] I have tried creating a repro with https://new.docusaurus.io.
- [x] I have read the console error message carefully (if applicable).

### Description

The main navigation dropdowns cannot be opened using the Enter key when a screen reader is active, preventing keyboard-only users from accessing sub-menu links.

### Steps to reproduce

1. On Windows, turn on the Narrator screen reader or any other screen reader.

2. Open the Docusaurus website in a browser.

3. Use the Tab key to navigate to a dropdown menu in the navbar, like the "Canary" version switcher or the "English"     
    language switcher.

4. Press the Enter key.

### Expected behavior

The dropdown menu should open, allowing the user to navigate the options.

### Actual behavior

 Nothing happens. The dropdown menu does not open when Enter is pressed.

### Your environment

- Public source code:
- Public site URL:
- Docusaurus version used:
- Environment name and version (e.g. Chrome 89, Node.js 16.4):
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):

### Self-service

- [x] I'd be willing to fix this bug myself.

**Comment by OfficialMutazAwad** [comment]
New reply to your ticket #12005

Title: [facebook/docusaurus] A11y: Navbar dropdowns do not open with screen reader (Issue #11501)

Message: Hello,we received your message and our support will reach out to you as soon as possible.

**Comment by Huzaif-nabi** [comment]
I'm working on this

**Comment by kostyngricuk** [comment]
Did you try to use space bar instead of Enter key ?

**Comment by slorber** [comment]
It works with/without VoiceOver on macOS, and we do have explicit code to handle keyboard "enter" being pressed:

```ts
onKeyDown={(e) => {
          if (e.key === 'Enter') {
            e.preventDefault();
            setShowDropdown(!showDropdown);
          }
        }}
```

If this doesn't work for your specific screen reader, I believe this should be considered a bug in your screen reader, and reported to them directly.

If this affacts a wide range of screen readers and systems, please let us know.
